libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteSession.kt

```diff
@@ -58,11 +60,23 @@ class ReplayRouteSession : MapboxNavigationObserver {
     private lateinit var replayRouteMapper: ReplayRouteMapper
     private var mapboxNavigation: MapboxNavigation? = null
     private var lastLocationEvent: ReplayEventUpdateLocation? = null
-    private var routesObserver: RoutesObserver? = null
-    private var currentRouteId: String? = null
+    private var currentRoute: NavigationRoute? = null
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        if (currentRoute?.id != routeProgress.navigationRoute.id) {
+            currentRoute = routeProgress.navigationRoute
+            onRouteChanged(routeProgress)
+        }
+    }
+
+    private val routesObserver = RoutesObserver { result ->
+        if (result.navigationRoutes.isEmpty()) {
+            mapboxNavigation?.resetReplayLocation()
+        }
```

Comment:
Route progress isn't triggered on routes cleanup, it it? I guess routes observer is the best place to cleanup current state?
```suggestion
        if (result.navigationRoutes.isEmpty()) {
            mapboxNavigation?.resetReplayLocation()
            currentRoute = null
            // + cleanup the point caches / streams
        }
```

Reply:
Yeah good idea